linker-interface/shared/src/main/scala/org/scalajs/linker/interface/ESFeatures.scala

```diff
@@ -26,36 +26,90 @@ import Fingerprint.FingerprintBuilder
  *    corresponding features if it supports them. Support for such options can
  *    be dropped in any subsequent version of the linker, including patch
  *    versions.
+ *  - Options whose name is of the form `avoidX` *hint* at the linker to avoid
+ *    the corresponding features *when it does not affect observable
+ *    semantics*. They are related to optimizations (for performance or code
+ *    size). The linker is free to ignore those options.
  */
 final class ESFeatures private (
-    /** Whether to use ECMAScript 2015 features, such as classes and arrow
-     *  functions.
-     */
-    val useECMAScript2015: Boolean,
-    /** EXPERIMENTAL: Whether to allow using ECMAScript `BigInt`s to implement
-     *  `Long`s.
-     */
-    val allowBigIntsForLongs: Boolean
+    _useECMAScript2015: Boolean,
+    _allowBigIntsForLongs: Boolean,
+    _avoidLetsAndsConsts: Boolean
 ) {
   import ESFeatures._
 
   private def this() = {
     this(
-        useECMAScript2015 = true,
-        allowBigIntsForLongs = false
+        _useECMAScript2015 = true,
+        _allowBigIntsForLongs = false,
+        _avoidLetsAndsConsts = true
     )
   }
 
+  /** Use ECMAScript 2015 features, such as classes and arrow functions.
+   *
+   *  Default: `true`
+   *
+   *  When `true`, the following behaviors are guaranteed:
+   *
+   *  - JavaScript classes are true `class`'es, therefore a) they can extend
+   *    native JavaScript `class`'es and b) they inherit static members from
+   *    their parent class.
+   *  - Throwable classes are proper JavaScript error classes, recognized as
+   *    such by debuggers.
+   *  - In Script (`NoModule`) mode, top-level exports are defined as `let`s
+   *    rather than `var`s, and behave as such.
```

Comment:
Are we actually testing that? (IIUC in practice this means they are not defined on the global object). If no, maybe we should?

Reply:
I have added a test for that in the first commit.